src/simplex.cpp

```diff
@@ -0,0 +1,466 @@
+// point while still being compatible with the floating point version.
+//
+// Warning: there is a patent on simplex noise for certain uses, which probably
+// doesn't include LED animations (no guarantee): US6867776. It is expected to
+// expire in early 2022.
+// See:
+// https://patents.stackexchange.com/questions/18573
+
+// Original author: Stefan Gustavson, converted to Go by Lars Pensjö, converted
+// to fixed-point Go and then to C++ by Ayke van Laethem.
+// https://github.com/larspensjo/Go-simplex-noise/blob/master/simplexnoise/simplexnoise.go
+// https://github.com/aykevl/ledsgo/blob/master/noise.go
+//
+// The code in this file has been placed in the public domain. You can do
+// whatever you want with it. Attribution is appreciated but not required.
+
+// Notation:
+// Every fixed-point calculation has a line comment saying how many bits in the
+// given integer are used for the fractional part. For example:
+//
+//     n := a + b // .12
+//
+// means the result of this operation has the floating point 12 bits from the
+// right. It can be converted to a floating point using:
+//
+//     nf := float64(n) / (1 << 12)
+
+FASTLED_NAMESPACE_BEGIN
+
+#define P(x) FL_PGM_READ_BYTE_NEAR(p + (x))
+
+// Permutation table. This is just a random jumble of all numbers.
+// This needs to be exactly the same for all instances on all platforms,
+// so it's easiest to just keep it as static explicit data.
+FL_PROGMEM static uint8_t const p[] = {
+    151, 160, 137, 91, 90, 15,
+    131, 13, 201, 95, 96, 53, 194, 233, 7, 225, 140, 36, 103, 30, 69, 142, 8, 99, 37, 240, 21, 10, 23,
+    190, 6, 148, 247, 120, 234, 75, 0, 26, 197, 62, 94, 252, 219, 203, 117, 35, 11, 32, 57, 177, 33,
+    88, 237, 149, 56, 87, 174, 20, 125, 136, 171, 168, 68, 175, 74, 165, 71, 134, 139, 48, 27, 166,
+    77, 146, 158, 231, 83, 111, 229, 122, 60, 211, 133, 230, 220, 105, 92, 41, 55, 46, 245, 40, 244,
+    102, 143, 54, 65, 25, 63, 161, 1, 216, 80, 73, 209, 76, 132, 187, 208, 89, 18, 169, 200, 196,
+    135, 130, 116, 188, 159, 86, 164, 100, 109, 198, 173, 186, 3, 64, 52, 217, 226, 250, 124, 123,
+    5, 202, 38, 147, 118, 126, 255, 82, 85, 212, 207, 206, 59, 227, 47, 16, 58, 17, 182, 189, 28, 42,
+    223, 183, 170, 213, 119, 248, 152, 2, 44, 154, 163, 70, 221, 153, 101, 155, 167, 43, 172, 9,
+    129, 22, 39, 253, 19, 98, 108, 110, 79, 113, 224, 232, 178, 185, 112, 104, 218, 246, 97, 228,
+    251, 34, 242, 193, 238, 210, 144, 12, 191, 179, 162, 241, 81, 51, 145, 235, 249, 14, 239, 107,
+    49, 192, 214, 31, 181, 199, 106, 157, 184, 84, 204, 176, 115, 121, 50, 45, 127, 4, 150, 254,
+    138, 236, 205, 93, 222, 114, 67, 29, 24, 72, 243, 141, 128, 195, 78, 66, 215, 61, 156, 180,
+};
+
+// A lookup table to traverse the simplex around a given point in 4D.
+// Details can be found where this table is used, in the 4D noise method.
+// TODO: This should not be required, backport it from Bill's GLSL code!
```

Comment:
I actually tried this, but the GLSL code is not faster.

1. I tried it in my Go implementation (running on my laptop), where it was actually a bit slower.
2. I tried roughly the same code on an ESP32, but it didn't affect the speed (the difference was well within the noise).

For those interested, see: https://gist.github.com/aykevl/dddc20182537f7c00f8b6358de97fc6d